pkgs/by-name/gh/ghciwatch/package.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, nix-update-script
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "ghciwatch";
+  version = "0.5.10";
+
+  src = fetchFromGitHub {
+    owner = "MercuryTechnologies";
+    repo = "ghciwatch";
+    rev = "v${version}";
+    hash = "sha256-6afUHLPrSWhgN5LA346tAZ1+gROr+i/ZyCNVnyCd5Tc=";
+  };
+
+  cargoHash = "sha256-og7S3W+DCBlFIvKLZghLT+msBLnS1o7Rea7v2VPsDYA=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.CoreServices
+  ];
+
+  # integration tests are not run but the macros need this variable to be set
```

Comment:
any reason why they can't be run? i think you might want the passthru.tests pattern if it's that you don't want to depend on ghc in your derivation.

Reply:
My opinion on this is: failures in the integration test suite are much more likely to indicate that _my_ code is broken upstream than it is to indicate that the packaging is broken.

`passthru.tests` would be fine but I also wouldn't want to be on the hook for keeping it working, y'know?

See: https://github.com/NixOS/nixpkgs/pull/308276#discussion_r1586536716